File: src/Domain/Entities/Booking.cs
using SkipperAgency.Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace SkipperAgency.Domain.Entities
{
    public class Booking
    {
        public int Id { get; set; }
        public int BoatId { get; set; }
        [ForeignKey(nameof(BoatId))]
        public virtual Boat Boat { get; set; }
        public string SkipperId { get; set; }
        [ForeignKey(nameof(SkipperId))]
        public virtual Skipper Skipper { get; set; }
        public string CharterId { get; set; }
        [ForeignKey(nameof(CharterId))]
        public virtual Charter Charter { get; set; }
        public DateTime BookedFrom { get; set; }
        public DateTime BookedTo { get; set; }
        public BookingStatusEnum Status { get; set; }
        public string BookingUrl { get; set; }
        public List<BookingHistory> BookingHistories { get; set; }
        public string OnBoardingLocation { get; set; }
        public string GuestName { get; set; }
        public string GuestEmail { get; set; }
        public int? GuestNationalityId { get; set; }
        [ForeignKey(nameof(GuestNationalityId))]
        public Country GuestNationality { get; set; }
        public string GuestMessage { get; set; }
        public bool GuestTos { get; set; }
        public int CrewSize { get; set; }
        public DateTime SkipperRequestTime { get; set; }
    }
}

File: src/WebUI/ClientApp/src/types/Booking.ts
import { ICountry } from './ICountry';
import { ILanguage } from './ILanguage';
import { ISkill } from './ISkill';
import { Boat } from "./Boat";
import { Skipper } from "./Skipper";
import { Charter } from "./Charter";

export interface Booking {
  id?: number;
  boatId?: number;
  boat?: Boat;
  skipperId?: string;
  skipper?: Skipper;
  charterId?: string;
  charter?: Charter;
  bookedFrom?: Date;
  bookedTo?: Date;
  status?: BookingStatusEnum;
  bookingURL?: string;
  bookingHistories?: BookingHistory[];
  skipperAction?: string;
  onboardingLocation?: string;
  guestName?: string;
  guestEmail?: string;
  guestTOS?: boolean
  guestNationality?: ICountry;
  crewSize?: number;
  guestMessege?: string;
  skipperRequestTime?: Date
}

export interface BookingHistory {
  id?: number;
  skipperId: number;
  skipper: Skipper;
  dateTime: Date | string;
  bookingRejected: BookingRejectionEnum;
}

export interface BookingsContext {
  bookings: Booking[];
  countries: ICountry[];
  getCountries: Function;
  setSkipperBookingsData: Function;
  setCharterBookingsData: Function;
  setBookingsData: Function;
}

export interface IEditBookingContext {
  showEditBooking: boolean;
  showNewBooking: boolean;
  booking: Booking;
  setBooking: Function;
  setShowEditBooking: Function;
  setShowNewBooking: Function;
}

export interface IGuestBookingContext {
  step: Number;
  booking: Booking;
  skills: ISkill[],
  languages: ILanguage[],
  setBooking: Function;
  fetchGuestBooking: Function;
  setStep: Function;
  getLanguages: Function;
  getSkills: Function;
}

export enum BookingStatusEnum {
  DeliveryFail = 0,
  SkipperRequestPending = 1,
  SkipperRequested = 2,
  SkipperAccepted = 3,
  SkipperDeclined = 4
}

export enum BookingRejectionEnum {
  SkipperRejected = 0,
  SkipperTimeOutExpired = 1
}

export enum SkipperActionEnum {
  Decline = 0,
  Accept = 1
}